src/main/java/org/apache/sysds/runtime/instructions/cp/UnaryCPInstruction.java

```diff
@@ -61,8 +63,9 @@ public static UnaryCPInstruction parseInstruction ( String str ) {
 			in.split(parts[1]);
 			out.split(parts[2]);
 			func = Builtin.getBuiltinFnObject(opcode);
-			
-			if( Arrays.asList(new String[]{"ucumk+","ucum*","ucumk+*","ucummin","ucummax","exp","log","sigmoid"}).contains(opcode) ){
+			Types.OpOp1 op_type = Types.OpOp1.valueOfByOpcode(opcode);
+			//if( Arrays.asList(new String[]{"ucumk+","ucum*","ucumk+*","ucummin","ucummax","exp","log","sigmoid","round"}).contains(opcode) )
+			if( Unary.isMultiThreadedOp(op_type)){
```

Comment:
This change here is twofold.
1. yes it is great that you changed it to not compare a list of strings, 
2. it seems like it allows fewer operations to enter the loop.

Reply:
Fewer? I have observed the opposite. Or maybe I don't understand what you mean.